packages/suite/src/components/wallet/Fees/FeeDetails.tsx

```diff
@@ -1,120 +1,291 @@
+        if (!ref.current) return;
+
+        resizeObserver(feeOptions, setColumns).observe(ref.current);
+
+        return () => resizeObserver(feeOptions, setColumns).disconnect();
+    }, [feeOptions.length, feeOptions, setColumns]);
 
     return (
         showFee && (
-            <>
-                <Item label={<Translation id="ESTIMATED_TIME" />}>
-                    {formatDuration(feeInfo.blockTime * selectedLevel.blocks * 60)}
-                </Item>
-
-                <Item label={<Translation id="TR_FEE_RATE" />}>
-                    {hasInfo ? transactionInfo.feePerByte : selectedLevel.feePerUnit}{' '}
-                    {getFeeUnits(networkType)}
-                    {hasInfo ? ` (${transactionInfo.bytes} B)` : ''}
-                </Item>
-            </>
+            <FeeCardsWrapper ref={ref} $columns={columns}>
+                {feeOptions &&
+                    feeOptions.map((fee, index) => (
+                        <FeeCard
+                            key={index}
+                            value={fee.value}
+                            setSelectedLevelOption={setSelectedLevelOption}
+                            isSelected={selectedLevelOption === fee.value}
+                            changeFeeLevel={changeFeeLevel}
+                            topLeftChild={
+                                <span data-testid={`fee-card/${fee.value}`}>{fee.label}</span>
+                            }
+                            topRightChild={
+                                <>~{formatDuration(feeInfo.blockTime * (fee?.blocks || 0) * 60)}</>
+                            }
+                            bottomLeftChild={
+                                <FiatValue
+                                    disableHiddenPlaceholder
+                                    amount={fee?.networkAmount || ''}
+                                    symbol={symbol}
+                                    showApproximationIndicator
+                                />
+                            }
+                            bottomRightChild={
+                                <>
+                                    {fee?.feePerUnit} {getFeeUnits(networkType)}
+                                    {hasInfo ? ` (${transactionInfo.bytes} B)` : ''}
+                                </>
+                            }
+                        />
+                    ))}
+            </FeeCardsWrapper>
         )
     );
 };
 
 const EthereumDetails = ({
-    networkType,
-    selectedLevel,
-    transactionInfo,
     showFee,
+    feeOptions,
+    setSelectedLevelOption,
+    selectedLevelOption,
+    changeFeeLevel,
+    symbol,
+    networkType,
 }: DetailsProps) => {
-    // States to remember the last known values of feeLimit and feePerByte when isComposedTx was true.
-    const [lastKnownFeeLimit, setLastKnownFeeLimit] = useState('');
-    const [lastKnownFeePerByte, setLastKnownFeePerByte] = useState('');
+    const isMainnet = symbol === 'eth';
```

Comment:
- Mainnet
	- The live blockchain where real transactions occur and actual value is transferred
	- Ethereum, BSC, POL
- Testnet 
	- A sandbox blockchain for testing and development, using tokens without real-world value
	- HOL, SEP
- Layer 2
	- Secondary solutions built on top of the mainnet to enhance scalability and reduce fees while still leveraging the main blockchain’s security
	- BASE, OP, ARB